sentry-samples/sentry-samples-android/src/main/java/io/sentry/samples/android/NetworkInterceptor.kt

```diff
@@ -0,0 +1,21 @@
+package io.sentry.samples.android
+
+import io.sentry.Sentry
+import io.sentry.SpanStatus
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class NetworkInterceptor : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+
+        // read transaction from the bound scope
+        val span = Sentry.getSpan()?.startChild("http.client", request.url().toString())
+
+        val response = chain.proceed(request)
```

Comment:
I believe we need a try catch here in case proceed actually throws.

Reply:
no, Retrofit implements its own Error handler, it bails to the `onFailure` callback, and this is part of our sample, not SDK